src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -1977,12 +1987,12 @@ UCS_PROFILE_FUNC_ALWAYS(ucs_status_t, uct_ib_mlx5_devx_reg_exported_key,
         goto out_umem_mr;
     }
 
-    status = uct_ib_mlx5_devx_reg_ksm_data_contig(md,
-                                                  &memh->mrs[UCT_IB_MR_DEFAULT],
-                                                  memh->address,
-                                                  (uint64_t)memh->address, 0, 0,
-                                                  "exported key", &cross_mr,
-                                                  &exported_lkey);
+    ucs_assert(!(memh->super.flags & UCT_IB_MEM_MULTITHREADED));
```

Comment:
BTW, do someone know should we care about that in case of `uct_ib_mlx5_devx_reg_exported_key` or not? This call also always assumes that MR was create by single thread.

Should we use `uct_ib_mlx5_devx_reg_ksm` wrapper? Or assert is OK for now?

Reply:
i'm not sure it is even supported. You can leave an assert for now